src/lib/y2partitioner/dialogs/btrfs_subvolume.rb

```diff
@@ -0,0 +1,94 @@
+require "yast"
+require "y2partitioner/dialogs/popup"
+
+module Y2Partitioner
+  module Dialogs
+    # Popup dialog to create a btrfs subvolume
+    class BtrfsSubvolume < Popup
+      attr_accessor :path
+      attr_accessor :nocow
+
+      def initialize
+        textdomain "storage"
+      end
+
+      def title
+        _("Add subvolume")
+      end
+
+      def contents
+        HVSquash(
+          VBox(
+            Left(SubvolumePath.new(self)),
+            Left(SubvolumeNocow.new(self))
+          )
+        )
+      end
+
+      def layout
+        VBox(
+          HSpacing(50),
+          Left(Heading(Id(:title), title)),
+          VStretch(),
+          VSpacing(1),
+          VBox(ReplacePoint(Id(:contents), Empty())),
+          VSpacing(1),
+          VStretch(),
+          ButtonBox(
+            PushButton(Id(:ok), Opt(:default), Yast::Label.OKButton),
+            PushButton(Id(:cancel), Yast::Label.CancelButton)
+          )
+        )
+      end
+    end
+  end
+
+  # Input field to set the subvolume path
+  class SubvolumePath < CWM::InputField
+    attr_reader :dialog
+
+    def initialize(dialog)
```

Comment:
I suggest to not call it dialog. Widgets should be independent on dialog that is using it. I suggest to call it something like generic data holder. So maybe something like `data` and document that requirement is to have methods `path/path=` . This way it is much easier in future to reuse this widget in different dialog and its dependency is much cleaner.